### PR TITLE
Email transcript use case

### PR DESCRIPTION
**Pull request by lucieyang1**
# Description
- Gets the messages from the session and formats it into a transcript string

Fixes # (issue)
- SessionId and SessionCode no longer get removed from the session storage, is that ok?

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Manually tested. Make sure to get the associated backend branch for code review.
image

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code

### CODE REVIEW

File: src/components/EndSession.jsx [hunk 1]
```diff
@@ -104,16 +91,47 @@ const Button = styled.button`
 `;
 
 function EndSession() {
+    const emailRef = useRef();
+    const sessionCode = sessionStorage.getItem("sessionCode");
+
+    useEffect(() => emailjs.init("OScF2lHq5ESl_o9lU"), []);
```
Comment (zjayee) [inline src/components/EndSession.jsx:97]: Are these api keys? If they are, they should be moved to the environment variables.

File: src/components/Chat.jsx [hunk 2]
```diff
@@ -200,8 +200,6 @@ function Chat() {
     // Clear the messages when the session ends
     setMessages([]);
     sessionStorage.removeItem("chatMessages");
-    sessionStorage.removeItem("sessionId")
-    sessionStorage.removeItem("sessionCode")
```
Comment (zjayee) [inline src/components/Chat.jsx:204]: If sessionId and sessionCode are not cleared here, there should be code in end session somewhere that clears them when the session ends. Perhaps when the user clicks on a return to home/start button. However, that may be out of the scope of this PR.

**Review by zjayee — COMMENTED** [review]
I've checked out the branch and the email transcript sending works on my end! Overall, coding is clear and follows style conventions. Just one question about the api.

File: src/components/EndSession.jsx (hunk 1 above)
Comment (zjayee) [reply src/components/EndSession.jsx:97]: Oh if it's a public api then it's fine. I thought it was private one.

[the remark this answers is not in the document]
Comment (sarinali) [reply src/components/EndSession.jsx:97]: I think we should use environment variable for this

**Review by sarinali — COMMENTED** [review]
LGTM, minor comment

File: src/components/EndSession.jsx [hunk 3]
```diff
@@ -104,16 +91,47 @@ const Button = styled.button`
...
+            const transcript = response.data;
+            console.log('Transcript:', transcript);
+
+            await emailjs.send(serviceId, templateId, {
+                recipient: emailRef.current.value,
+                message: transcript
+            });
+            alert("Email successfully sent. Check inbox.");
+        } catch (error) {
+            alert("Oops! Something went wrong while trying to send the email. Please make sure there are messages in the conversation before sending.");
```
Comment (3milyfz) [inline src/components/EndSession.jsx:122]: Can you change the alerts to mui alerts to stay consistent with the ones in my pull request? 
- This is the documentation: https://mui.com/material-ui/react-alert/
- You can reference my pr to see how to make them disappear automatically after a certain amount of time so users don't need to close it

**Review by 3milyfz — CHANGES_REQUESTED** [review]
Good job! Just a minor comment on use of alert.

File: src/components/EndSession.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucieyang1) [reply src/components/EndSession.jsx:97]: I tried putting the key into an .env file, but for some reason, it shows up as undefined when I try to run the code. There might be something wrong with what I'm doing, though I searched google and it seems like someone had the same issue 2 years ago, that went unresolved: https://stackoverflow.com/questions/66491790/why-can-t-i-send-messages-with-emailjs-when-hiding-my-keys-in-env

**Review by 3milyfz — APPROVED** [review]
LGTM!

**Pull request merged**